sympy/physics/optics/gaussopt.py

```diff
@@ -582,9 +588,10 @@ def radius(self):
         return self.z*(1 + (self.z_r/self.z)**2)
 
     @property
-    def w(self):
+    def w_z(self):
```

Comment:
Why are you changing the name here? And is it also the radius at `1/e^2` intensity just like `w0` or did you intend to remove that part of the definition for `w`?

Reply:
Yes , in all online documentations `w_z` is the symbol used for the property spot size/ beam width .Secondly it was confusing when I saw the code because of same name of the input parameter w (beam waist) and w (spot size) function .

Also `w_0 `and` w_z` are basically the same properties but at different z ( waist distance), `w_0`  at the minimum and w_z at a general z . They both are `1/e^2` of the intensity values at the current instant . That's why I mentioned earlier the doc-string is partially correct and added peak value and axial value of intensity to it.

Reply:
I don't find the word "spot" in SymPy -- where is the spot size talked about?

Reply:
> I don't find the word "spot" in SymPy -- where is the spot size talked about?

Spot size is just another name for beam width ,it is a quantity I referred here If my above explanation wasn't clear . 
As far as `'w'` and `'w-z'` are concerned -
1. `'w_z'` is the symbol used in all online documentations ( see links in PR description) so anyways it must be changed .
2. Currently  `'w'` is referring to beam waist ( input argument to `BeamParameter(Expr)` ) and also as function `w(self)`( function returning the beam width/spot size ) .Code wise it will work but this will be confusing for a user to understand.

What I have done is replace function `w() ` with `w_z()` ( radius of beam at a position z along axis ) .This makes it inline with all online documentation and also with current function `w_0()` (radius of beam at z=0 or minimal beam radius)

Reply:
Any thoughts @smichr ?

Reply:
@smichr I am confident of this change .I'll resolve this conversation unless you have some doubts . Thanks

Reply:
> 1. `'w_z'` is the symbol used in all online documentations ( see links in PR description) so anyways it must be changed .

What equations in the provided links have `w_z` in them? I find `w(z)` a lot, but not `w_z`. I might be missing it though.

Reply:
Yeah I have used ` w_z ` to represent `w(z)` . Is there any other standard way to write a function which varies with a particular parameter (like w does here with axial distance z) which should be used here ?

Reply:
I hope with this [comment](https://github.com/sympy/sympy/pull/22969#discussion_r808958096) I have established that 2 element names (one parameter and one function) are same so one should be changed, then what the name whether `wz` or `w_z` is trivial and can be decided .The links were just for justification of conceptual changes made to the code ,not for the name precisely .

Reply:
Since `z` is a property of the object, then just `w` isn't exactly inappropriate either. So no, I don't believe you have established that it should be changed. If we had this discussion when this was first created it would be a different situation, but I don't see a reason to break the API here. Please take into consideration the inconvenience this brings to users.

Reply:
I was under the impression that this will bring clarity to users ,for example the if we consider w only then this test is confusing -
`assert p.w == w*sqrt(l**2*z**2/(pi**2*w**4) + 1)`
since w on lhs is not same as w on rhs  [comment](https://github.com/sympy/sympy/pull/22969#discussion_r808958096). 
If you still think `w` is appropriate I have no objection to changing `w_z` back to it .

Reply:
```python
    z, l, w = symbols('z l r', positive=True)
    p = BeamParameter(l, z, w=w)
```
Isn't the problem here? Since the python variable `w` is actually `Symbol('r')`. If we would print the rhs and lhs (instead of look at the python code) the ambiguity would disappear right?

If anything, the input parameter should have been `w_0` as in
```python
    p = BeamParameter(l, z, w_0='Symbol('w_0'))
```
But again, I don't think it is worth bothering the user with such a change. Changing the python variable name `w` in the test to make it clearer should not be a problem (since it does not change anything for the user)

Reply:
> Isn't the problem here? Since the python variable `w` is actually `Symbol('r').

Thanks for the sleuthing. I'll revert `w_z` back to `w` in a commit .I wouldn't bother to change the input parameter in it unless you want it to be .

Reply:
I have made the change . Other than this, are there any other potential changes that you suggest can be made. Else this pr seems correct now !